### PR TITLE
Close #247: Update firmware automatically

### LINKED ISSUE #247: Update firmware automatically

### PR DESCRIPTION
**Pull request by TheSomeMan**

### CODE REVIEW

File: tests/test_json_helper/test_json_helper.cpp [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+          "\"created_at\": \"2021-07-19T11:38:50Z\",\n"
+          "\"published_at\": \"2021-07-19T14:08:10Z\"\n"
+          "}";
+    ASSERT_EQ(string("MDc6UmVsZWFzZTQ2NDE0Nzcy"), json_helper_get_by_key_wrapper(p_json, "node_id"));
+    ASSERT_EQ(string("v1.6.0"), json_helper_get_by_key_wrapper(p_json, "tag_name"));
+    ASSERT_EQ(string("2021-07-19T11:38:50Z"), json_helper_get_by_key_wrapper(p_json, "created_at"));
+    ASSERT_EQ(string("2021-07-19T14:08:10Z"), json_helper_get_by_key_wrapper(p_json, "published_at"));
+}
+
+TEST_F(TestJsonHelper, test_json_get_by_key_2) // NOLINT
```
Comment (ojousima) [inline tests/test_json_helper/test_json_helper.cpp:75]: What's the difference between this and test 1?

**Review by ojousima — APPROVED** [review]
Thanks, please see comments

File: main/adv_post.c [hunk 2]
```diff
@@ -265,58 +293,107 @@ adv_post_retransmit_advs(const adv_report_table_t *p_reports, const bool flag_co
...
 
-        if (0 != g_adv_reports_buf.num_of_advs)
+static void
+adv_post_task(void)
+{
+    esp_log_level_set(TAG, ESP_LOG_INFO);
+
+    if (!os_signal_register_cur_thread(g_p_adv_post_sig))
+    {
+        LOG_ERR("%s failed", "os_signal_register_cur_thread");
```
[the remark this answers is not in the document]
Comment (TheSomeMan) [reply main/adv_post.c:336]: thanks, yes, it'd be better to exit from the thread

File: tests/test_json_helper/test_json_helper.cpp (hunk 1 above)
Comment (TheSomeMan) [reply tests/test_json_helper/test_json_helper.cpp]: I've renamed these tests

**Pull request merged**